### PR TITLE
Docs: Replaces Agent with Alloy

### PR DESCRIPTION
**Pull request by imatwawana**
Replaces references to Grafana Agent with Grafana Alloy

### CODE REVIEW

File: docs/sources/introduction/_index.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ In addition to Grafana, Grafana Labs also provides the following open source pro
...
 
-**Grafana Agent:** Grafana Agent is a vendor-neutral, batteries-included telemetry collector with configuration inspired by Terraform. It is designed to be flexible, performant, and compatible with multiple ecosystems such as Prometheus and OpenTelemetry. For more information about Grafana Agent, refer to [Grafana Agent documentation](/docs/agent/latest/).
+**Grafana Alloy:** Grafana Alloy is a vendor-neutral, batteries-included telemetry collector with configuration inspired by Terraform. It is designed to be flexible, performant, and compatible with multiple ecosystems such as Prometheus and OpenTelemetry. For more information about Grafana Alloy, refer to [Grafana Alloy documentation](https://grafana.com/docs/alloy/latest/).
```
Comment (clayton-cornell) [inline docs/sources/introduction/_index.md:87]:
```suggestion
**Grafana Alloy:** Grafana Alloy is a flexible, high performance, vendor-neutral distribution of the [OpenTelemetry][] (OTel) Collector.
It's fully compatible with the most popular open source observability standards such as OpenTelemetry (OTel) and Prometheus.
For more information about Grafana Alloy, refer to the [Grafana Alloy documentation](https://grafana.com/docs/alloy/latest/).
```

File: docs/sources/setup-grafana/configure-grafana/_index.md [hunk 2]
```diff
@@ -2009,7 +2009,7 @@ Depending on the value of `sampler_type`, the sampler configuration parameter ca
 
 When `sampler_type` is `remote`, this specifies the URL of the sampling server. This can be used by all tracing providers.
 
-Use a sampling server that supports the Jaeger remote sampling API, such as jaeger-agent, jaeger-collector, opentelemetry-collector-contrib, or [Grafana Agent](/oss/agent/).
+Use a sampling server that supports the Jaeger remote sampling API, such as jaeger-agent, jaeger-collector, opentelemetry-collector-contrib, or [Grafana Alloy](https://grafana.com/oss/alloy-opentelemetry-collector/).
```
Comment (clayton-cornell) [inline docs/sources/setup-grafana/configure-grafana/_index.md:2012]: Interesting link target for Alloy. It's not a bad link, but unexpected vs a link to the Alloy docs at https://grafana.com/docs/alloy/latest/

File: docs/sources/fundamentals/intro-to-prometheus/index.md [hunk 3]
```diff
@@ -106,17 +106,17 @@ The following image shows the two metrics associated with the endpoint. The HELP
...
 
-Metrics data is only one type of telemetry data; the other kinds are logs and traces. Using Grafana Agent can be a great option to send telemetry data because as you scale your observability practices to include logs and traces, which Grafana Agent also supports, you've got a solution already in place.
+Metrics data is only one type of telemetry data; the other kinds are logs and traces. Using Grafana Alloy can be a great option to send telemetry data because as you scale your observability practices to include logs and traces, which Grafana Alloy also supports, you've got a solution already in place.
 
-The following image illustrates how Grafana Agent works as an intermediary between 'MyApp' and Grafana Cloud.
+The following image illustrates how Grafana Alloy works as an intermediary between 'MyApp' and Grafana Cloud.
 
-{{< figure src="/media/docs/grafana/intro-prometheus/grafana-agent.png" max-width="750px" caption="Grafana Agent" >}}
+{{< figure src="/media/docs/alloy/flow-diagram-small-alloy.png" max-width="750px" alt="Grafana Alloy" caption="Grafana Alloy" >}}
```
Comment (clayton-cornell) [inline docs/sources/fundamentals/intro-to-prometheus/index.md:115]: Do we need the max width arg here?

**Review by clayton-cornell — APPROVED** [review]
Looks OK. Some suggestions for aligning the description of Alloy and a couple non-blocking questions.

File: docs/sources/setup-grafana/configure-grafana/_index.md (hunk 2 above)
Comment (imatwawana) [reply docs/sources/setup-grafana/configure-grafana/_index.md:2012]: I was just replacing like with like. Do you want a docs link there?

Comment (imatwawana) [reply docs/sources/setup-grafana/configure-grafana/_index.md:2012]: I was replacing like with like. Do you want a docs link there?

File: docs/sources/fundamentals/intro-to-prometheus/index.md (hunk 3 above)
Comment (imatwawana) [inline docs/sources/fundamentals/intro-to-prometheus/index.md:115]:
```suggestion
{{< figure src="/media/docs/alloy/flow-diagram-small-alloy.png" alt="Grafana Alloy" caption="Grafana Alloy" >}}
```

File: docs/sources/setup-grafana/configure-grafana/_index.md (hunk 2 above)
Comment (clayton-cornell) [reply docs/sources/setup-grafana/configure-grafana/_index.md:2012]: Either/or. It's not a bad link. It references the Alloy marketing info on the main site. Leave it.. merge it :-)

**Pull request merged**